test_scenarios/kubernetes/3idxc3shc1cm1lm1dep-pvc/splunk-indexer-statefulset-persistent.yaml

```diff
@@ -0,0 +1,73 @@
+apiVersion: apps/v1beta1
+kind: StatefulSet
+metadata:
+  name: indexer
+  #namespace: splunk
+spec:
+  replicas: 3
+  serviceName: indexer
+  volumeClaimTemplates:
+  - metadata:
+      name: splunk-idxcluster-data
+    spec:
+      accessModes: [ "ReadWriteOnce" ]
+      resources:
+        requests:
+          storage: 100Gi
+  - metadata: 
+      name: splunk-idxcluster-config
+    spec:
+      accessModes: [ "ReadWriteOnce" ]
+      resources:
+        requests:
+          storage: 10Gi
+  template:
+    metadata:
+      labels:
+        app: splunk
+        role: splunk_indexer
+        tier: idxcluster
+    spec:
+      #serviceAccountName: splunk
+      dnsPolicy: ClusterFirst
+      dnsConfig:
+        searches:
+          - indexer.splunk.svc.cluster.local
+          - search.splunk.svc.cluster.local
```

Comment:
FWIW, this is only required if use short names (e.g.`indexer-0`) in the `env` settings below. It may be preferable to omit this and use a ConfigMap to apply those settings (ref: https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#define-container-environment-variables-using-configmap-data) allowing the user to have more flexibly across namespaced clusters.

Reply:
Totally agree. This was definitely hard-coded for simplicity.  Did you build with fqdn in the env vars and have it build ok?

Reply:
We built with fqdn, e.g.
```
            - name: SPLUNK_INDEXER_CLUSTER_MASTER
              value: "splunk-indexer-master.default.svc.cluster.local"
```

Reply:
TBH, though... I didn't try that short hand, so not sure if it would work.